icu4c/source/common/localebuilder.cpp

```diff
@@ -401,6 +401,24 @@ Locale makeBogusLocale() {
   return bogus;
 }
 
+void LocaleBuilder::copyExtensionsFrom(const Locale& src, UErrorCode& errorCode)
+{
+    if (U_FAILURE(errorCode)) { return; }
+    LocalPointer<icu::StringEnumeration> iter(src.createKeywords(errorCode));
+    if (U_FAILURE(errorCode) || iter.isNull() || iter->next(nullptr, errorCode) == nullptr) {
```

Comment:
I didn't see one. I modified the code to use count() instead of next(), and to pass this enumeration into the copy helper so we need not get it twice.